### ISSUE TITLE
Selecting multiple devices causes a failure

### ISSUE THREAD

**Issue by jordan-romeroporter-earnin** [lyndsey-ferguson/fastlane-plugin-test_center#179]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

Multi-Scan

When selecting the device type and having multiple Simulators running in parallel, it results in a crash/multi-scan failure.
 
Here is my Fastfile:

```
  multi_scan(
    workspace: "MyProject.xcworkspace",
    try_count: 3,
    scheme: 'MyProject-Dev',
    parallel_testrun_count: 1,
    only_testing: values[:only_testing],
    include_simulator_logs: false, 
    result_bundle:true,
    output_style: "rspec",
    output_directory: "./build/DerivedData/test_output",
    derived_data_path: "./build/DerivedData",
    device:'iPhone 8 (13.2.2)',
    output_types: "junit"
  )
```
##### Complete output when running fastlane, including the stack trace and command used

```
The test runner encountered an error (Failed to prepare device 'iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70162821190540>' for impending launch. (Underlying error: Unable to boot the Simulator. launchd failed to respond.))
```

<details>
  <pre>[19:59:02]: Deleting iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70162821190540>
Traceback (most recent call last):
	65: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<main>'
	64: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'
	63: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/bin/fastlane:23:in `<top (required)>'
	62: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
	61: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	60: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
	59: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	58: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	57: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	56: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	55: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	54: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	53: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	52: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	51: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	50: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	49: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	48: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	47: from Fastfile:266:in `block (2 levels) in parsing_binding'
	46: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	45: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
	44: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
	43: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	42: from Fastfile:398:in `block in parsing_binding'
	41: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	40: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	39: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	38: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	37: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
	36: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	35: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
	34: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:31:in `run'
	33: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:82:in `run_summary'
	32: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:82:in `each'
	31: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:83:in `block in run_summary'
	30: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/other_action.rb:23:in `method_missing'
	29: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	28: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	27: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	26: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
	25: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	24: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
	23: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/tests_from_junit.rb:5:in `run'
	22: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/tests_from_junit.rb:5:in `new'
	21: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:15:in `initialize'
	20: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	19: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	18: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	17: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	16: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:16:in `block in initialize'
	15: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:16:in `new'
	14: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:29:in `initialize'
	13: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	12: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	11: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	10: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	 9: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:30:in `block in initialize'
	 8: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:30:in `new'
	 7: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:47:in `initialize'
	 6: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	 5: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	 4: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	 3: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	 2: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:48:in `block in initialize'
	 1: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:48:in `new'
/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:87:in `initialize': undefined method `value' for nil:NilClass (NoMethodError)
	65: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<main>'
	64: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'
	63: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/bin/fastlane:23:in `<top (required)>'
	62: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
	61: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	60: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
	59: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	58: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	57: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	56: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	55: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	54: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	53: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	52: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	51: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	50: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	49: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	48: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	47: from Fastfile:266:in `block (2 levels) in parsing_binding'
	46: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	45: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
	44: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
	43: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	42: from Fastfile:398:in `block in parsing_binding'
	41: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	40: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	39: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	38: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	37: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
	36: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	35: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
	34: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:31:in `run'
	33: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:82:in `run_summary'
	32: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:82:in `each'
	31: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:83:in `block in run_summary'
	30: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/other_action.rb:23:in `method_missing'
	29: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	28: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	27: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	26: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
	25: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	24: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
	23: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/tests_from_junit.rb:5:in `run'
	22: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/tests_from_junit.rb:5:in `new'
	21: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:15:in `initialize'
	20: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	19: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	18: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	17: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	16: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:16:in `block in initialize'
	15: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:16:in `new'
	14: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:29:in `initialize'
	13: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	12: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	11: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	10: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	 9: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:30:in `block in initialize'
	 8: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:30:in `new'
	 7: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:47:in `initialize'
	 6: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `each'
	 5: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	 4: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/xpath.rb:68:in `each'
	 3: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rexml/element.rb:927:in `block in each'
	 2: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:48:in `block in initialize'
	 1: from /Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:48:in `new'
/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/junit_helper.rb:87:in `initialize': [!] undefined method `value' for nil:NilClass (NoMethodError)</pre>
</details>

### Environment

<details>
  <pre>[INSERT OUTPUT HERE]</pre>
</details>

**Comment by lyndsey-ferguson** [comment]
Can you provide the logs and use the `--verbose` flag? I want to see what is happening before this crash.

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson when I run with --verbose && --capture_output all I get is this:

```
#!/bin/bash --login -o pipefail
bundle exec fastlane test_center --verbose --capture_output
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [⠧] 🚀 [✔] 🚀 
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=60):
Unable to boot the Simulator.
launchd failed to respond.

Exited with code exit status 1
```

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson looks like --capture_output was problematic, below is what it produced:

```
[worker 2] DEBUG [2019-12-12 22:34:07.09]: Deleting xcresults:
[worker 2] DEBUG [2019-12-12 22:34:07.09]: Restarting Simulator 8D440462
[worker 2] INFO [2019-12-12 22:34:07.44]: Starting scan #1 with 3 tests for batch #2.
[worker 2] Successfully loaded Appfile at path '/Users/distiller/project/fastlane/Appfile'
[worker 2] -------
[worker 2] Successfully loaded Appfile at path '/Users/distiller/project/fastlane/Appfile'
[worker 2] -------
[worker 2] DEBUG [2019-12-12 22:34:07.45]: retrying_scan #update_scan_options
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting workspace to MyProject.xcworkspace
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting scheme to MyProject-Dev
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting only_testing to ["MyProjectUITests/SignUpTests/testSignUp", "MyProjectUITests/SignUpTests/testSignupWithAccountSelection", "MyProjectUITests/SignUpTests/testSignupMarketplace"]
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting include_simulator_logs to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting result_bundle to true
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting output_style to rspec
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting output_directory to /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting derived_data_path to /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/derived_data_path20191212-15670-8o39kk-worker-2
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting output_types to junit
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting skip_detect_devices to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting reset_simulator to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting reinstall_app to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting xctestrun to ./build/DerivedData/Build/Products/MyProject-Dev_iphonesimulator13.2-x86_64.xctestrun
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting clean to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting open_report to false
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting output_files to report.junit
[worker 2] DEBUG [2019-12-12 22:34:07.45]: 	Setting buildlog_path to output/buildlogs/scan/worker-2-logs
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting should_zip_build_products to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting use_clang_report_name to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting disable_concurrent_testing to true
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting skip_build to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting build_for_testing to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting xcargs to  -parallel-testing-enabled NO
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting slack_use_webhook_configured_username_and_icon to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting slack_username to fastlane
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting slack_icon_url to https://s3-eu-west-1.amazonaws.com/fastlane.tools/fastlane.png
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting skip_slack to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting slack_only_on_failure to false
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting destination to ["platform=iOS Simulator,id=8D440462-03D5-4CB1-91F8-29A6322F8B2A"]
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting xcodebuild_command to env NSUnbufferedIO=YES xcodebuild
[worker 2] DEBUG [2019-12-12 22:34:07.46]: 	Setting fail_build to true
[worker 2] 
[worker 2] +------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
[worker 2] |                                                                                                                                Summary for scan 2.137.0                                                                                                                                |
[worker 2] +------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
[worker 2] | workspace                                      | MyProject.xcworkspace                                                                                                                                                                                                               |
[worker 2] | scheme                                         | MyProject-Dev                                                                                                                                                                                                                       |
[worker 2] | only_testing                                   | ["MyProjectUITests/SignUpTests/testSignUp", "MyProjectUITests/SignUpTests/testSignupWithAccountSelection", "MyProjectUITests/SignUpTests/testSignupMarketplace"] |
[worker 2] | include_simulator_logs                         | false                                                                                                                                                                                                                                 |
[worker 2] | result_bundle                                  | true                                                                                                                                                                                                                                  |
[worker 2] | output_style                                   | rspec                                                                                                                                                                                                                                 |
[worker 2] | output_directory                               | /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2                                                                                                                                                     |
[worker 2] | derived_data_path                              | /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/derived_data_path20191212-15670-8o39kk-worker-2                                                                                                                                      |
[worker 2] | output_types                                   | junit                                                                                                                                                                                                                                 |
[worker 2] | skip_detect_devices                            | false                                                                                                                                                                                                                                 |
[worker 2] | force_quit_simulator                           | false                                                                                                                                                                                                                                 |
[worker 2] | reset_simulator                                | false                                                                                                                                                                                                                                 |
[worker 2] | reinstall_app                                  | false                                                                                                                                                                                                                                 |
[worker 2] | xctestrun                                      | ./build/DerivedData/Build/Products/MyProject-Dev_iphonesimulator13.2-x86_64.xctestrun                                                                                                                                               |
[worker 2] | clean                                          | false                                                                                                                                                                                                                                 |
[worker 2] | open_report                                    | false                                                                                                                                                                                                                                 |
[worker 2] | output_files                                   | report.junit                                                                                                                                                                                                                          |
[worker 2] | buildlog_path                                  | output/buildlogs/scan/worker-2-logs                                                                                                                                                                                                   |
[worker 2] | should_zip_build_products                      | false                                                                                                                                                                                                                                 |
[worker 2] | use_clang_report_name                          | false                                                                                                                                                                                                                                 |
[worker 2] | disable_concurrent_testing                     | true                                                                                                                                                                                                                                  |
[worker 2] | skip_build                                     | false                                                                                                                                                                                                                                 |
[worker 2] | xcargs                                         |  -parallel-testing-enabled NO                                                                                                                                                                                                         |
[worker 2] | slack_use_webhook_configured_username_and_icon | false                                                                                                                                                                                                                                 |
[worker 2] | slack_username                                 | fastlane                                                                                                                                                                                                                              |
[worker 2] | slack_icon_url                                 | https://s3-eu-west-1.amazonaws.com/fastlane.tools/fastlane.png                                                                                                                                                                        |
[worker 2] | skip_slack                                     | false                                                                                                                                                                                                                                 |
[worker 2] | slack_only_on_failure                          | false                                                                                                                                                                                                                                 |
[worker 2] | xcodebuild_command                             | env NSUnbufferedIO=YES xcodebuild                                                                                                                                                                                                     |
[worker 2] | fail_build                                     | true                                                                                                                                                                                                                                  |
[worker 2] | xcode_path                                     | /Applications/Xcode-11.2.1.app                                                                                                                                                                                                        |
[worker 2] | build_for_testing                              | false                                                                                                                                                                                                                                 |
[worker 2] +------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
[worker 2] 
[worker 2] INFO [2019-12-12 22:34:07.52]: $ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -destination 'platform=iOS Simulator,id=8D440462-03D5-4CB1-91F8-29A6322F8B2A' -derivedDataPath '/var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/derived_data_path20191212-15670-8o39kk-worker-2' -resultBundlePath '/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/MyProject-Dev.test_result' -disable-concurrent-testing -xctestrun './build/DerivedData/Build/Products/MyProject-Dev_iphonesimulator13.2-x86_64.xctestrun'  -parallel-testing-enabled NO -only-testing:MyProjectUITests/SignUpTests/testSignUp -only-testing:MyProjectUITests/SignUpTests/testSignupAccountSelection -only-testing:MyProjectUITests/SignUpTests/testSignupMarketplace test-without-building | tee '/Users/distiller/project/output/buildlogs/scan/worker-2-logs/MyProject-MyProject-Dev.log' | xcpretty --test --report junit --output '/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/report.junit' --report junit --output '/var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/junit_report20191212-22336-iftj14' 
[worker 2] INFO [2019-12-12 22:34:07.52]: ▸ Loading...
[worker 2] INFO [2019-12-12 22:34:24.41]: ▸ 2019-12-12 22:34:24.413 xcodebuild[22356:82836]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
[worker 2] INFO [2019-12-12 22:34:24.49]: ▸ /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/MyProject-Dev.test_result/Staging/1_Test/Diagnostics/MyProjectUITests-A8858A28-7403-46D6-96FA-E82DDAEBBD23/MyProjectUITests-161730C8-4046-4A4D-B0EC-5F9933D89997/Session-MyProjectUITests-2019-12-12_223424-eEkqHi.log
[worker 2] INFO [2019-12-12 22:34:24.49]: ▸ 2019-12-12 22:34:24.415 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: (C35C519C-EE2B-4DE1-BD20-1F110837705F) Beginning test session MyProjectUITests-C35C519C-EE2B-4DE1-BD20-1F110837705F at 2019-12-12 22:34:24.415 with Xcode 11B500 on target <DVTiPhoneSimulator: 0x7fc44686e080> {
[worker 2] INFO [2019-12-12 22:34:24.49]: ▸ 		SimDevice: iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70348912136940> (8D440462-03D5-4CB1-91F8-29A6322F8B2A, iOS 13.2, Shutdown)
[worker 2] INFO [2019-12-12 22:34:24.49]: ▸ } (13.2.2 (17B102))
[worker 2] INFO [2019-12-12 22:37:44.30]: ▸ 2019-12-12 22:37:44.306 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 199.915 elapsed -- Testing started completed.
[worker 2] INFO [2019-12-12 22:37:44.30]: ▸ 2019-12-12 22:37:44.307 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
[worker 2] INFO [2019-12-12 22:37:44.31]: ▸ 2019-12-12 22:37:44.307 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 199.915 sec, +199.915 sec -- end
[worker 2] INFO [2019-12-12 22:37:44.31]: ▸ Testing failed:
[worker 2] INFO [2019-12-12 22:37:44.31]: ▸ 	MyProjectUITests:
[worker 2] INFO [2019-12-12 22:37:44.31]: ▸ 		The test runner encountered an error (Failed to prepare device 'iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70348912136940>' for impending launch. (Underlying error: Unable to boot the Simulator. launchd failed to respond.))
[worker 2] INFO [2019-12-12 22:37:44.32]: ▸ ** TEST EXECUTE FAILED **
[worker 2] 2019-12-12 22:34:24.413 xcodebuild[22356:82836]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
[worker 2] /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/MyProject-Dev.test_result/Staging/1_Test/Diagnostics/MyProjectUITests-A8858A28-7403-46D6-96FA-E82DDAEBBD23/MyProjectUITests-161730C8-4046-4A4D-B0EC-5F9933D89997/Session-MyProjectUITests-2019-12-12_223424-eEkqHi.log
[worker 2] 2019-12-12 22:34:24.415 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: (C35C519C-EE2B-4DE1-BD20-1F110837705F) Beginning test session MyProjectUITests-C35C519C-EE2B-4DE1-BD20-1F110837705F at 2019-12-12 22:34:24.415 with Xcode 11B500 on target <DVTiPhoneSimulator: 0x7fc44686e080> {
[worker 2] 		SimDevice: iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70348912136940> (8D440462-03D5-4CB1-91F8-29A6322F8B2A, iOS 13.2, Shutdown)
[worker 2] } (13.2.2 (17B102))
[worker 2] 2019-12-12 22:37:44.306 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 199.915 elapsed -- Testing started completed.
[worker 2] 2019-12-12 22:37:44.307 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
[worker 2] 2019-12-12 22:37:44.307 xcodebuild[22356:82450] [MT] IDETestOperationsObserverDebug: 199.915 sec, +199.915 sec -- end
[worker 2] Testing failed:
[worker 2] 	MyProjectUITests:
[worker 2] 		The test runner encountered an error (Failed to prepare device 'iPhone 8 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70348912136940>' for impending launch. (Underlying error: Unable to boot the Simulator. launchd failed to respond.))
[worker 2] 
[worker 2] ** TEST EXECUTE FAILED **
[worker 2] ERROR [2019-12-12 22:37:47.60]: Exit status: 65
[worker 2] DEBUG [2019-12-12 22:37:47.61]: Batch #2 incrementing retry count to 1
[worker 2] DEBUG [2019-12-12 22:37:47.61]: Scan unable to test for batch #2
[worker 2] ERROR [2019-12-12 22:37:47.61]: 
[worker 2] Error building the application. See the log above.
[worker 2] ["/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane_core/lib/fastlane_core/ui/interface.rb:163:in `build_failure!'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/scan/lib/scan/error_handler.rb:31:in `handle_build_error'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/scan/lib/scan/runner.rb:70:in `block in test_app'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane_core/lib/fastlane_core/command_executor.rb:89:in `execute'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/scan/lib/scan/runner.rb:61:in `test_app'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/scan/lib/scan/runner.rb:22:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:59:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:45:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/test_batch_worker.rb:18:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/parallel_test_batch_worker.rb:49:in `block in run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/parallel_test_batch_worker.rb:45:in `fork'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/parallel_test_batch_worker.rb:45:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:143:in `block in run_test_batches'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each_with_index'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `run_test_batches'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:53:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-test_center-3.8.15/lib/fastlane/plugin/test_center/actions/multi_scan.rb:29:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'", "Fastfile:399:in `block in parsing_binding'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/lane.rb:33:in `call'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'", "Fastfile:267:in `block (2 levels) in parsing_binding'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/lane.rb:33:in `call'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/runner.rb:45:in `execute'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/gems/fastlane-2.137.0/bin/fastlane:23:in `<top (required)>'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'", "/Users/distiller/project/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<top (required)>'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli/exec.rb:74:in `load'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli/exec.rb:74:in `kernel_load'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli/exec.rb:28:in `run'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli.rb:465:in `exec'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli.rb:27:in `dispatch'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/cli.rb:18:in `start'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/exe/bundle:30:in `block in <top (required)>'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'", "/Library/Ruby/Gems/2.6.0/gems/bundler-2.0.2/exe/bundle:22:in `<top (required)>'", "/usr/local/bin/bundle:23:in `load'", "/usr/local/bin/bundle:23:in `<main>'"]
[worker 2] DEBUG [2019-12-12 22:37:47.61]: All tests passed for batch 2? false
DEBUG [2019-12-12 22:49:27.44]: Collating results for all batches
DEBUG [2019-12-12 22:49:27.44]: ReportCollator collating
DEBUG [2019-12-12 22:49:27.44]: Collating junit report files ["/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/report.junit", "/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-1/report.junit"]
DEBUG [2019-12-12 22:49:27.44]: collate_junit_reports with ["/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/report.junit", "/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-1/report.junit"]
DEBUG [2019-12-12 22:49:27.46]: > collating last report file /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-1/report.junit
DEBUG [2019-12-12 22:49:27.46]: < collating last report file /Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-1/report.junit
DEBUG [2019-12-12 22:49:27.47]: Collating test_result bundles ["/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-2/MyProject-Dev.test_result", "/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests-batch-1/MyProject-Dev-1.test_result"]
DEBUG [2019-12-12 22:49:27.47]: result bundles are of format version 3
INFO [2019-12-12 22:49:27.51]: xcrun xcresulttool merge /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/MyProject-Dev.test_result20191212-15670-basv8e.xcresult /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/MyProject-Dev-1.test_result20191212-15670-1oxbl4l.xcresult --output-path /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/d20191212-15670-p5il6b/MyProject-Dev.test_result
INFO [2019-12-12 22:49:27.51]: $ xcrun xcresulttool merge /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/MyProject-Dev.test_result20191212-15670-basv8e.xcresult /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/MyProject-Dev-1.test_result20191212-15670-1oxbl4l.xcresult --output-path /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/d20191212-15670-p5il6b/MyProject-Dev.test_result
INFO [2019-12-12 22:49:30.47]: ▸ [v3] Merged to: /var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/d20191212-15670-p5il6b/MyProject-Dev.test_result
INFO [2019-12-12 22:49:30.84]: Finished collating test_result bundle to '/Users/distiller/project/build/DerivedData/test_output/MyProjectUITests/MyProject-Dev.test_result'
DEBUG [2019-12-12 22:49:31.55]: Results for each test run: [true, false]
WARN [2019-12-12 22:49:31.55]: Lane Context:
INFO [2019-12-12 22:49:31.55]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios test_center", :ORIGINAL_DEFAULT_KEYCHAIN=>"\"""}
INFO [2019-12-12 22:49:31.55]: Successfully generated documentation at path '/Users/distiller/project/fastlane/README.md'
```

**Comment by lyndsey-ferguson** [comment]
This gives me some things to think about. Did you see logs from `[worker 1]`? Was it able to spin up the Simulator? I can imagine that it may have had problems and what you saw was the _first_ failure.

**Comment by lyndsey-ferguson** [comment]
Also, what happens if you remove the `parallel_testrun_count` option?

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson I did see logs from worker-1, the first time all the tests failed, then the second run all the tests passed. However prior to worker-1, it tried to run an xcodebuild command and had "problems". Below is what the code looked like.

```
INFO [2019-12-12 22:23:50.34]: $ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace MyProject.xcworkspace -scheme MyProject-Dev -destination 'platform=iOS Simulator,id=35C35F56-63D0-4EFF-885C-FFB0D9C08A42' -derivedDataPath './build/DerivedData' -resultBundlePath './build/DerivedData/test_output/MyProject-Dev.test_result' -only-testing:MyProjectUITests/SignUpTests/testSignup -only-testing:MyprojectUITests/SignUpTests/testSignupMarketplace -only-testing:MyProjectUITests/SignUpTests/testSignUp -only-testing:MyProjectUITests/LoginTest/testText -only-testing:MyProjectUITests/UITests/testTextLocal -only-testing:MyProjectUITests/Group1Tests/testText build-for-testing | tee '/Users/distiller/project/output/buildlogs/scan/MyProject-MyProject-Dev.log' | xcpretty --test --report junit --output '/Users/distiller/project/build/DerivedData/test_output/report.junit' --report junit --output '/var/folders/z7/8rr0l_45719dk91y30bbhvw40000gn/T/junit_report20191212-15670-1gm80ui' 
INFO [2019-12-12 22:23:50.36]: ▸ Loading...
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 0 |
| Number of failures | 0 |
+--------------------+---+

DEBUG [2019-12-12 22:33:54.76]: After building, found xctestrun files ["./build/DerivedData/Build/Products/MyProject-Dev_iphonesimulator13.2-x86_64.xctestrun"] (choosing 1st)
DEBUG [2019-12-12 22:33:54.76]: Removing report files generated by the build
DEBUG [2019-12-12 22:33:54.76]:   ./build/DerivedData/test_output/report.junit
DEBUG [2019-12-12 22:33:54.76]: Removing pre-existing test_result bundles: 
DEBUG [2019-12-12 22:33:54.76]:   /Users/distiller/project/build/DerivedData/test_output/MyProject-Dev.test_result
DEBUG [2019-12-12 22:33:54.76]: Fetching available simulator devices
DEBUG [2019-12-12 22:33:55.42]: Fetching available simulator devices
DEBUG [2019-12-12 22:33:55.70]: Cloning simulator
DEBUG [2019-12-12 22:33:56.05]: Cloning simulator
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=60):
Unable to boot the Simulator.
launchd failed to respond.
INFO [2019-12-12 22:34:07.08]: Starting test run 1
INFO [2019-12-12 22:34:07.08]: Starting test run 2
```

**Comment by jordan-romeroporter-earnin** [comment]
I'm not sure, I can try now. But a follow up question is, what would be the benefit of not adding it? I'm hoping to parallelize the automation process via this.

**Comment by lyndsey-ferguson** [comment]
A benefit for me to see if this is a problem with parallelization or with the simulators.

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson when I set it to 1, it works great - and I'm currently using it on our CI tool. If that provides any clarity

**Comment by lyndsey-ferguson** [comment]
The next step may be to get the sim that matches, and try and run a simple command in parallel on each to see if there we can boot the simulator.

**Comment by jordan-romeroporter-earnin** [comment]
How can I help?

**Comment by lyndsey-ferguson** [comment]
I won't be able to write it right now as i am at work. My idea in pseudocode:
1. get the UDID of the Simulator that matches your selected sim
2. try and boot that with xcrun simctl boot <UDID>
3. Does that fail?
4. If that does not fail: make clones of that simulator
5. For each one, try booting them, does that fail?

There could be a problem with the original sim, in which case I'd reach out to CircleCI. If there are problems with the clones, that would require more debugging. 

This is the first time I've seen this particular issue, and I know that other people are using parallel (me too).

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson so this is actually a limitation within CircleCI, so not a bug with your library. I'll figure out a work around - thank you for working through this with me.

**Comment by lyndsey-ferguson** [comment]
@jordan-romeroporter-earnin how do you know that? Is there a link to docs on that? I _thought_ that other people with CircleCI were using this. If this is documented, I'd like to warn users about this, and even alert people in the plugin if they try this.

**Comment by jordan-romeroporter-earnin** [comment]
@lyndsey-ferguson  I'm working with a support engineer that told me:

> Relating to the question about multiple simulators running in a single container you are right, this is a limitation in the way our infrastructure is set up to allocate the resources that one average job needs. The reason why is because splitting those resources across multiple jobs running in a single instance would result in far fewer resources being available per job, which could cause slowness if not outright failures best practice is instead to spin up multiple separate instances that run simultaneously.

**Comment by lyndsey-ferguson** [comment]
Ok, I'll add a warning to those people that use Circle CI. For this case then, I suggest using the actions that are in this plugin to:
1. decide how many batches of test runs you want to run.
2. create a CircleCI job for each batch
3. use the `tests_from_xctestrun` action to get the complete list of tests
4. using that list, split that into batches ([example code](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/lib/fastlane/plugin/test_center/helper/test_collector.rb#L121).
5. run a given batch for each CircleCI job by sending it to `multi_scan` using the `only_testing` parameter.

If you can create multiple CircleCI jobs in your CI, you can basically achieve the same effect.

**Comment by lyndsey-ferguson** [comment]
Later on, I'll create an Issue that references this one stating that CircleCI does not support parallel and to make it error out early.

**Comment by lyndsey-ferguson** [comment]
I am leaving this open as a reminder to create the other Issue

**Comment by lyndsey-ferguson** [comment]
Closed as I created #183
